core/frontend/src/main/scala/io/udash/properties/seq/ForwarderSeqProperty.scala

```diff
@@ -1,19 +1,132 @@
+      listeners.clear()
+      onListenerInit()
+      originListenerRegistration = origin.listen(originListener)
+    }
+    if (originStructureListenerRegistration == null || !originStructureListenerRegistration.isActive()) {
+      structureListeners.clear()
+      originStructureListenerRegistration = origin.listenStructure(originStructureListener)
+    }
+  }
+
+  protected def killOriginListeners(): Unit = {
+    if (originListenerRegistration != null && listeners.isEmpty) {
+      originListenerRegistration.cancel()
+      originListenerRegistration = null
+    }
+    if (originStructureListenerRegistration != null && listeners.isEmpty) {
+      originStructureListenerRegistration.cancel()
+      originStructureListenerRegistration = null
+    }
+  }
+
+  override def listenStructure(structureListener: (Patch[ElemType]) => Any): Registration = {
+    initOriginListeners()
+    structureListeners += structureListener
+    wrapListenerRegistration(new JsArrayRegistration(structureListeners, structureListener))
+  }
+
+  override def listen(valueListener: (Seq[B]) => Any, initUpdate: Boolean = false): Registration = {
+    initOriginListeners()
+    wrapListenerRegistration(super.listen(valueListener, initUpdate))
+  }
+
+  override def listenOnce(valueListener: (Seq[B]) => Any): Registration = {
+    initOriginListeners()
+    wrapListenerRegistration(super.listenOnce(valueListener))
+  }
+
+  protected def wrapListenerRegistration(reg: Registration): Registration = new Registration {
+    override def restart(): Unit = {
+      initOriginListeners()
+      reg.restart()
+    }
+
+    override def cancel(): Unit = {
+      reg.cancel()
+      killOriginListeners()
+    }
+
+    override def isActive(): Boolean =
+      reg.isActive()
+  }
 }
 
-trait ForwarderSeqProperty[A, +ElemType <: Property[A]]
-  extends ForwarderReadableSeqProperty[A, ElemType] with ForwarderProperty[Seq[A]] with SeqProperty[A, ElemType] {
+trait ForwarderWithLocalCopy[A, B, ElemType <: ReadableProperty[B], OrigType <: ReadableProperty[A]]
+  extends ForwarderReadableSeqProperty[A, B, ElemType, OrigType] {
+
+  import js.JSConverters._
+
+  protected var transformedElements: js.Array[ElemType] = js.Array()
+
+  protected def loadFromOrigin(): Seq[B]
+  protected def elementsFromOrigin(): Seq[ElemType]
+  protected def transformPatchAndUpdateElements(patch: Patch[OrigType]): Patch[ElemType]
+
+  override def get: Seq[B] = {
+    if (originListenerRegistration == null || !originListenerRegistration.isActive()) loadFromOrigin()
+    else transformedElements.map(_.get).toSeq
+  }
+
+  override def elemProperties: Seq[ElemType] = {
+    if (originListenerRegistration == null || !originListenerRegistration.isActive()) elementsFromOrigin()
+    else transformedElements.toSeq
+  }
+
+  override protected def onListenerInit(): Unit = {
+    val fromOrigin = elementsFromOrigin().toJSArray
+    if (transformedElements.map(_.id) != fromOrigin.map(_.id)) {
```

Comment:
This can be done without materializing the collections:
```scala
!(transformedElements.iterator.map(_.id) sameElements fromOrigin.iterator.map(_.id))
```